components/oauth/org.wso2.carbon.identity.discovery/src/main/java/org/wso2/carbon/identity/discovery/DefaultOIDCProcessor.java

```
@@ -0,0 +1,73 @@
+ */
+package org.wso2.carbon.identity.discovery;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.base.ServerConfigurationException;
+import org.wso2.carbon.identity.discovery.builders.DefaultOIDProviderRequestBuilder;
+import org.wso2.carbon.identity.discovery.builders.OIDProviderRequestBuilder;
+import org.wso2.carbon.identity.discovery.builders.ProviderConfigBuilder;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DefaultOIDCProcessor implements OIDCProcessor {
+    private static Log log = LogFactory.getLog(DefaultOIDCProcessor.class);
+    private static DefaultOIDCProcessor defaultOidcProcessor = new DefaultOIDCProcessor();
+
+
+    private DefaultOIDCProcessor() {
+        if (log.isDebugEnabled()) {
+            log.debug("Initializing DefaultOIDCProcessor for OpenID connect discovery processor.");
+        }
+    }
+
+    public static DefaultOIDCProcessor getInstance() {
+        return defaultOidcProcessor;
+    }
+
+    public OIDProviderConfigResponse getResponse(HttpServletRequest request, String tenant) throws
+            OIDCDiscoveryEndPointException, ServerConfigurationException {
+        OIDProviderRequestBuilder requestBuilder = new DefaultOIDProviderRequestBuilder();
+        OIDProviderRequest requestObject = requestBuilder.buildRequest(request, tenant);
+        ProviderConfigBuilder responseBuilder = new ProviderConfigBuilder();
+        return responseBuilder.buildOIDProviderConfig(requestObject);
+    }
+
+    public int handleError(OIDCDiscoveryEndPointException error) {
+        if (log.isDebugEnabled()) {
+            log.debug(error);
+        }
+        String errorCode = error.getErrorCode();
+        if (errorCode.equals(OIDCDiscoveryEndPointException.ERROR_CODE_NO_OPENID_PROVIDER_FOUND)) {
+
+        } else if (errorCode.equals(OIDCDiscoveryEndPointException.ERROR_CODE_INVALID_REQUEST)) {
+
```

The error codes are not specified in the spec. Should return the proper error code. This was left as a future work. I have added logs at https://github.com/wso2/carbon-identity/pull/1694. Shall I add @TODO